lib/common/strings.c

```diff
@@ -148,6 +150,123 @@ crm_parse_int(const char *text, const char *default_text)
     return (int) result;
 }
 
+/*!
+ * \internal
+ * \brief Scan a double-precision floating-point value from a string
+ *
+ * \param[in]      text         The string to parse
+ * \param[out]     result       Parsed value on success, or -1 on error
+ * \param[in]      default_text Default string to parse if \p text is
+ *                              \c NULL
+ * \param[out]     end_text     If not \c NULL, where to store a pointer
+ *                              to the position immediately after the
+ *                              value
+ *
+ * \return Standard Pacemaker return code (\c pcmk_rc_ok on success,
+ *         \c pcmk_rc_no_output on \c NULL \p result,
+ *         \c pcmk_rc_no_input on no input string,
+ *         \c pcmk_rc_unknown_format on failed string conversion due to
+ *         invalid input, \c pcmk_rc_after_range on arithmetic overflow,
+ *         \c pcmk_rc_before_range on arithmetic underflow, or \c errno
+ *         from \c strtod() on other parse errors)
+ */
+int
+pcmk__scan_double(const char *text, double *result, const char *default_text,
+                  char **end_text)
```

Comment:
In this iteration, I made pcmk__scan_double internal but non-static. I'm undecided on whether it should be static... It does all the work, so it's a helper function in that regard, but it accepts one argument (end_text) that pcmk__parse_double() does not.

The int parsing functions get by just fine without exposing scan_ll()'s end_text argument outside the file, however. So maybe this ought to be static.

We're accepting a pointer to hold the result and returning success/failure because that's consistent with the usual scan interface. pcmk__parse_double() is designed as a more user-friendly wrapper that returns the parsed value and stores the return code in a pointer. The parsed value is going to be used a lot more often as a function argument than the success/failure code.